### PR TITLE
Separate link/s for data privacy statement and imprint + markdown

### PR DESCRIPTION
**Pull request by MatthiasReumann**
- Added `texts` endpoint
- Added Texts Admin Tab
- Some UI changes 

### UI

image

On the mobile devices this merges with the new side nav:

image

Admin Tab:

image

### INSERT Statement for About, Imprint, and Data Privacy (Needs to be inserted after deployment)
```
insert into info_pages
    (id, created_at, updated_at, name, raw_content, type)
values
    (1, NOW(), NOW(), 'Privacy', '# Datenschutzerklärung

Für die Technische Universität München ist Datenschutz ein wichtiges Anliegen. Wir möchten, dass Sie wissen,
wann wir welche Daten speichern und wie wir sie verwenden. Personenbezogene Daten werden auf diesem zentralen
Web-Server der Universität (www.tum.de) nur im technisch notwendigen Umfang erhoben. Die Datenverarbeitung
unterliegt den geltenden datenschutzrechtlichen Bestimmungen, insbesondere der Datenschutz-Grundverordnung
(DSGVO) dem Bayerischen Datenschutzgesetz (BayDSG) und dem Telemediengesetz (TMG).Nachfolgend informieren wir Sie über Art, Umfang und Zweck der Erhebung und Verwendung personenbezogener Daten.
Diese Informationen können jederzeit von unserer Webseite abgerufen werden.

## Verwendete personenbezogene Daten
Bei einer Anmeldung werden die Folegenden personenbezogenen Daten erhoben:

- Anonymer personenbezogener Identifier ("Obfuscated ID")
- LRZ ID
- Matrikelnummer
- Name
- Kursanmeldungen aus TUMOnline

Die zum Login verwendeten Daten werden ausschließlich zur identifizierung an den TUM-weiten identity Provider
gesendet und im Anschluss verworfen.

## Protokollierung
Wenn Sie diese Internetseite aufrufen, übermitteln Sie über Ihren Internetbrowser Daten an unseren Webserver.
Die folgenden Daten werden während einer laufenden Verbindung zur Kommunikation zwischen Ihrem Internetbrowser
und unserem Webserver temporär in einer Logdatei aufgezeichnet:

- IP-Adresse des anfragenden Rechners</li>
- Datum und Uhrzeit des Zugriffs</li>
- Name, URL und übertragene Datenmenge der abgerufenen Datei
- Zugriffsstatus (angeforderte Datei übertragen, nicht gefunden etc.)
- Erkennungsdaten des verwendeten Browser- und Betriebssystems (sofern vom anfragenden Webbrowser übermittelt)
 -Webseite, von der aus der Zugriff erfolgte (sofern vom anfragenden Webbrowser übermittelt)

Die Verarbeitung der Daten in dieser Logdatei geschieht wie folgt:

- Die Logeinträge werden kontinuierlich automatisch ausgewertet, um Angriffe auf die Webserver erkennen und entsprechend reagieren zu können.
- In Einzelfällen, d.h. bei gemeldeten Störungen, Fehlern und Sicherheitsvorfällen, erfolgt eine manuelle Analyse.
- Logeinträge, die älter als sieben Tage sind, werden durch Kürzung der IP-Adresse anonymisiert.
- Die anonymisierten Logs werden zur Erstellung von Zugriffsstatistiken verwendet. Die hierfür eingesetzte Software wird lokal von uns betrieben.

## Cookies
Um den Funktionsumfang unseres Internetangebotes zu erweitern und die Nutzung für Sie komfortabler zu gestalten,
verwenden wir zum Teil so genannte „Cookies". Mit Hilfe dieser Cookies können bei dem Aufruf unserer Webseite
Daten auf Ihrem Rechner gespeichert werden. Sie können das Speichern von Cookies jedoch deaktivieren oder Ihren
Browser so einstellen, dass Cookies nur für die Dauer der jeweiligen Verbindung zum Internet gespeichert werden.
Hierdurch könnte allerdings der Funktionsumfang unseres Angebotes eingeschränkt werden.


Für einige Funktionen (gespeicherte Einstellungen wie Lautstärke oder ausgeblendete Kurse) werden außerdem Daten
im Webstorage Ihres Browsers gespeichert. Diese werden nicht an unsere Server übertragen.', 1),
    (2, NOW(), NOW(), 'Imprint', '# Impressum

## Herausgeber
- Technische Universität München
- Arcisstraße 21
- 80333 München
- Telefon: +4989289-01
- poststelle@tum.de

## Vertretungsberechtigt
Die Technische Universität München wird gesetzlich vertreten durch den Präsidenten, Prof. Dr. Thomas F. Hofmann.

## Zuständige Aufsichtsbehörde
Bayerisches Staatsministerium für Wissenschaft und Kunst

## Umsatzsteueridentifikationsnummer
DE811193231 (gemäß § 27a Umsatzsteuergesetz)

## Inhaltlich verantwortlich
- Stefan Schwalb
- Boltzmannstr. 3
- 85748 Garching
- E-Mail: schwalb(at)in.tum.de

Vorlesungsinhalte geben die Auffassungen und Erkenntnisse der entsprechenden Vortragenden wieder.

## Nutzungsbedingungen
Texte, Bilder, Grafiken sowie die Gestaltung dieses Webauftritts können dem Urheberrecht unterliegen.
Als Privatperson dürfen Sie urheberrechtlich geschütztes Material zum privaten und sonstigen eigenen Gebrauch im Rahmen des § 53 UrhG verwenden. Eine Vervielfältigung oder Verwendung urheberrechtlich geschützten Materials dieser Seiten oder Teilen davon in anderen elektronischen oder gedruckten Publikationen und deren Veröffentlichung ist nur mit unserer Einwilligung gestattet. Diese Einwilligung erteilen auf Anfrage die für den Inhalt Verantwortlichen. Der Nachdruck und die Auswertung von Pressemitteilungen und Reden sind mit Quellenangabe allgemein gestattet.
Weiterhin können Texte, Bilder, Grafiken und sonstige Dateien ganz oder teilweise dem Urheberrecht Dritter unterliegen. Auch über das Bestehen möglicher Rechte Dritter geben Ihnen die für den Inhalt Verantwortlichen nähere Auskünfte.

## Haftungsausschluss
Alle in diesem Webauftritt bereitgestellten Informationen haben wir nach bestem Wissen und Gewissen erarbeitet und geprüft. Eine Gewähr für die jederzeitige Aktualität, Richtigkeit, Vollständigkeit und Verfügbarkeit der bereit gestellten Informationen können wir allerdings nicht übernehmen. Ein Vertragsverhältnis mit den Nutzern des Webauftritts kommt nicht zustande.
Wir haften nicht für Schäden, die durch die Nutzung dieses Webauftritts entstehen. Dieser Haftungsausschluss gilt nicht, soweit die Vorschriften des § 839 BGB (Haftung bei Amtspflichtverletzung) einschlägig sind. Für etwaige Schäden, die beim Aufrufen oder Herunterladen von Daten durch Schadsoftware oder der Installation oder Nutzung von Software verursacht werden, übernehmen wir keine Haftung.

## Links
Von unseren eigenen Inhalten sind Querverweise („Links“) auf die Webseiten anderer Anbieter zu unterscheiden. Durch diese Links ermöglichen wir lediglich den Zugang zur Nutzung fremder Inhalte nach § 8 Telemediengesetz. Bei der erstmaligen Verknüpfung mit diesen Internetangeboten haben wir diese fremden Inhalte daraufhin überprüft, ob durch sie eine mögliche zivilrechtliche oder strafrechtliche Verantwortlichkeit ausgelöst wird. Wir können diese fremden Inhalte aber nicht ständig auf Veränderungen überprüfen und daher auch keine Verantwortung dafür übernehmen. Für illegale, fehlerhafte oder unvollständige Inhalte und insbesondere für Schäden, die aus der Nutzung oder Nichtnutzung von Informationen Dritter entstehen, haftet allein der jeweilige Anbieter der Seite.', 1),
    (3, NOW(), NOW(), 'About', 'Willkommen bei TUM-Live, dem Livestreaming und VoD Service der Rechnerbetriebsgruppe an der Fakultät für Informatik und Mathematik der Technischen Universität München. Bei Fragen, Kritik, Lob oder technischen Support stehen wir Ihnen gern unter live@rbg.tum.de zur Verfügung.

## Übertragung von Veranstaltungen

Wir übertragen gern alle Veranstaltungen, die sich an TUM-Studierende oder Mitglieder richten. Hier gibt es zwei Möglichkeiten:

**1. Die Veranstaltung findet in einem von uns ausgestatteten Raum statt:**

- [MW0001](https://nav.tum.sexy/room/5510.EG.001),
- [MW2001](https://nav.tum.sexy/room/5510.02.001),
- [MI HS 1](https://nav.tum.sexy/room/5602.EG.001),
- [MI HS 2](https://nav.tum.sexy/room/5604.EG.011),
- [MI HS 3](https://nav.tum.sexy/room/5606.EG.011),
- [00.08.038 (Seminarraum)](https://nav.tum.sexy/room/5608.EG.038),
- [00.13.009A (Seminarraum)](https://nav.tum.sexy/room/5613.EG.009A),
- [00.07.014 (Seminarraum)](https://nav.tum.sexy/room/5607.EG.014),
- [Interims I (101)](https://nav.tum.sexy/room/5620.01.101),
- [Interims I (102)](https://nav.tum.sexy/room/5620.01.102)

In diesem Fall wird die Veranstaltung automatisch übertragen und aufgezeichnet. Sie bekommen vor Semesterbeginn eine Benachrichtigung von uns und können einstellen, ob die Veranstaltung übertragen werden soll, welche Features aktiviert werden und wer sie sehen darf.
Voraussetzung hierfür ist, dass die Termine als Vorlesungstermine eingetragen sind und Sie in TUMOnline als Kontaktperson für den Kurs vermerkt sind. Sollte hier etwas nicht zu Ihrer Veranstaltungen passen, können Sie uns per Mail Bescheid geben, damit wir den Kurs eintragen können.

**2. Sie streamen die Veranstaltung selbst:**

Sollten Sie die Veranstaltung von einem anderen Ort aus abhalten, können Sie sie selbst bei uns einspeisen. Das geht mit Software wie [OBS](https://obsproject.com/de) oder [Zoom](https://support.zoom.us/hc/en-us/articles/115001777826-Live-Streaming-Meetings-or-Webinars-Using-a-Custom-Service).
', 1);```

### CODE REVIEW

**Review by joschahenningsen — APPROVED** [review]
Looks great! 
I think the dsgvo requires us to have the imprint and data privacy statements visible (reachable with only one click) but I'd still keep this as is for now. 

`texts` could be refactored to `infoPages` but that does not bother me too much :)

File: api/texts.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+type markdownTextRoutes struct {
+	dao.DaoWrapper
+}
+
+type updateTextDao struct {
+	Name       string `json:"name"`
+	RawContent string `json:"content"`
+	Type       uint   `json:"type"`
```
Comment (joschahenningsen) [inline api/texts.go]: Can you create a type for this? e.g. like here: https://github.com/joschahenningsen/TUM-Live/blob/c8c2d8b540889f9587e71074e2b37eb8082e7551/model/file.go#L9

File: api/texts.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		api.PUT("/texts/:id", routes.updateText)
+	}
+}
+
+type markdownTextRoutes struct {
+	dao.DaoWrapper
+}
+
+type updateTextDao struct {
+	Name       string `json:"name"`
```
Comment (joschahenningsen) [inline api/texts.go]: Does the name serve any purpose?

Comment (MatthiasReumann) [reply api/texts.go]: The name is used on the admin tab to differentiate between texts, but other than that, no.

**Comment by MatthiasReumann** [comment]
> Looks great! I think the dsgvo requires us to have the imprint and data privacy statements visible (reachable with only one click) but I'd still keep this as is for now.
> 
> `texts` could be refactored to `infoPages` but that does not bother me too much :)

`infoPages` is definitely a better name, hence, I renamed `texts` to `infoPages`.

**Comment by joschahenningsen** [comment]
Thanks a ton!

**Pull request merged**